src/deploy/functions/build.ts

```diff
@@ -0,0 +1,397 @@
+  platform?: "gcfv1" | "gcfv2";
+
+  // Necessary for the GCF API to determine what code to load with the Functions Framework.
+  // Will become optional once "run" is supported as a platform
+  entryPoint: string;
+
+  // The services account that this function should run as. Has no effect for a Run service.
+  // defaults to the GAE service account when a function is first created as a GCF gen 1 function.
+  // Defaults to the compute service account when a function is first created as a GCF gen 2 function
+  // or when using Cloud Run.
+  serviceAccount: ServiceAccount | null;
+
+  // defaults to ["us-central1"], overridable in firebase-tools with
+  //  process.env.FIREBASE_FUNCTIONS_DEFAULT_REGION
+  region?: string[];
+
+  // Firebase default of 80. Cloud default of 1
+  concurrency?: number | Expression<number> | null;
+
+  // Default of 256
+  availableMemoryMb?: number | Expression<number> | null;
+
+  // Default of 60
+  timeoutSeconds?: number | Expression<number> | null;
+
+  // Default of 1000
+  maxInstances?: number | Expression<number> | null;
+
+  // Default of 0
+  minInstances?: number | Expression<number> | null;
+
+  vpc?: VpcSettings | null;
+  ingressSettings?: "ALLOW_ALL" | "ALLOW_INTERNAL_ONLY" | "ALLOW_INTERNAL_AND_GCLB" | null;
+
+  environmentVariables?: Record<string, string | Expression<string>>;
+  labels?: Record<string, string | Expression<string>>;
+};
+
+interface ParamBase<T extends string | number | boolean> {
+  // name of the param. Will be exposed as an environment variable with this name
+  param: string;
+
+  // A human friendly name for the param. Will be used in install/configure flows to describe
+  // what param is being updated. If omitted, UX will use the value of "param" instead.
+  label?: string;
+
+  // A long description of the parameter's purpose and allowed values. If omitted, UX will not
+  // provide a description of the parameter
+  description?: string;
+
+  // Default value. If not provided, a param must be supplied.
+  default?: T | Expression<T>;
+
+  // default: false
+  immutable?: boolean;
+}
+
+interface StringParam extends ParamBase<string> {
+  type?: "string";
+}
+
+type Param = StringParam;
+
+function isMemoryOption(value: backend.MemoryOptions | any): value is backend.MemoryOptions {
+  return value == null || [128, 256, 512, 1024, 2048, 4096, 8192].includes(value);
+}
+
+/** Converts a build specification into a Backend representation, with all Params resolved and interpolated */
+// TODO(vsfan): resolve build.Params
+// TODO(vsfan): handle Expression<T> types
+export function discoverParams(build: Build): backend.Backend {
+  const bkEndpoints: Array<backend.Endpoint> = [];
+  for (const endpointId of Object.keys(build.endpoints)) {
+    const endpoint = build.endpoints[endpointId];
+
+    let regions = endpoint.region;
+    if (typeof regions === "undefined") {
+      regions = ["functionsDefaultRegion"];
```

Comment:
Let's use the actual functionsDefaultRegion

Reply:
Done.